contracts/seed/Seed.sol

```diff
@@ -0,0 +1,376 @@
+        uint16  vestingDuration;
+        uint16  vestingCliff;
+        uint16  daysClaimed;
+        uint256 totalClaimed;
+        address recipient;
+    }
+
+    /**
+      * @dev                          Initialize Seed.
+      * @param _beneficiary           The address of the beneficiary revieving the fee.
+      * @param _admin                 The address of the admin of this contract.
+      * @param _seedToken             The address of the token being distributed.
+      * @param _fundingToken          The address of the token being exchanged for seed token.
+      * @param _successMinimumAndCap  Array of the minimum distribution threshold and
+                                      the highest possible amount to be raised.
+      * @param _startTime             Distribution start time in unix timecode.
+      * @param _endTime               Distribution end time in unix timecode.
+      * @param _vestingDuration       Vesting period duration.
+      * @param _vestingCliff          Cliff duration.
+      * @param _isWhitelisted         Set to true if only whitelisted adresses are allowed to participate.
+      * @param _fee                   Success fee as %
+    */
+    function initialize(
+            address _beneficiary,
+            address _admin,
+            address _seedToken,
+            address _fundingToken,
+            uint[] memory    _successMinimumAndCap,
+            uint    _price,
+            uint    _startTime,
+            uint    _endTime,
+            uint16  _vestingDuration,
+            uint16  _vestingCliff,
+            bool    _isWhitelisted,
+            uint8   _fee
+    ) public initializer {
+        beneficiary     = _beneficiary;
+        admin           = _admin;
+        successMinimum  = _successMinimumAndCap[0];
+        cap             = _successMinimumAndCap[1];
+        price           = _price;
+        startTime       = _startTime;
+        endTime         = _endTime;
+        vestingDuration = _vestingDuration;
+        vestingCliff    = _vestingCliff;
+        isWhitelisted   = _isWhitelisted;
+        seedToken       = IERC20(_seedToken);
+        fundingToken    = IERC20(_fundingToken);
+        fee             = _fee;
+        closed          = false;
+        minimumReached  = false;
+    }
+
+    /**
+      * @dev                     Buy seed tokens.
+      * @param _amount           The amount of tokens to buy.
+    */
+    function buy(uint256 _amount) public protected checked {
+        require((fundingToken.balanceOf(address(this)) + _amount) <= cap, "Seed: amount exceeds contract sale cap");
+        require(fundingToken.transferFrom(msg.sender, address(this), _amount), "Seed: no tokens");
+
+        if (fundingToken.balanceOf(address(this)) >= successMinimum) {
+            minimumReached = true;
+        }
+
+        fundingTokensPerAddress[msg.sender] = _amount;
+
+        uint feeAmount = (_amount.mul(uint(PPM))).mul(fee).div(100000000); // 100000000 = PPM * 100
```

Comment:
can you please move this mafic value to the variable? maybe we can call it PPM100 or PPMBASE